src/main/java/sample/BadCredentialsException.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample;
+
+public class BadCredentialsException extends RuntimeException {
```

Comment:
Let's add the minimal amount of code to get things working for this initial iteration. Please remove this and instead the `Filter` can set the status if it catches an exception from the `AuthenticationManager`.